packages/react/src/components/FileUploader/FileUploaderDropContainer.js

```diff
@@ -0,0 +1,152 @@
+import React, { useCallback, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { settings } from 'carbon-components';
+import { keys, matches } from '../../internal/keyboard';
+import uniqueId from '../../tools/uniqueId';
+
+const { prefix } = settings;
+
+export default function FileUploaderDropContainer(props) {
+  const inputRef = useRef();
+  const {
+    accept,
+    className,
+    id,
+    disabled,
+    labelText,
+    multiple,
+    name,
+    onAddFiles,
+    role,
+    tabIndex,
+    ...other
+  } = props;
+  const uid = useRef(uniqueId()).current;
+  const [isActive, setActive] = useState(false);
+  const classes = classNames(`${prefix}--file__drop-container`, {
+    [`${prefix}--file__drop-container--drag-over`]: isActive,
+    [className]: className,
+  });
+  const handleChange = useCallback(
+    evt => {
+      onAddFiles(evt, {
+        addedFiles:
+          evt.type === 'drop'
+            ? [...evt.dataTransfer.files]
+            : [...evt.target.files],
+      });
+    },
+    [onAddFiles]
+  );
+  return (
+    <div
+      className={`${prefix}--file`}
+      onDragOver={evt => {
+        evt.stopPropagation();
+        evt.preventDefault();
+        setActive(true);
+        evt.dataTransfer.dropEffect = 'copy';
+      }}
+      onDragLeave={evt => {
+        evt.stopPropagation();
+        evt.preventDefault();
+        setActive(false);
+        evt.dataTransfer.dropEffect = 'move';
+      }}
+      onDrop={evt => {
+        evt.stopPropagation();
+        evt.preventDefault();
+        setActive(false);
+        handleChange(evt);
+      }}>
+      <label
+        className={`${prefix}--file-browse-btn`}
+        htmlFor={id || uid}
```

Comment:
```suggestion
        htmlFor={id || uid.current}
```

Per suggestion above